src/imagery/i.eodag/i.eodag.py

```diff
@@ -300,82 +396,199 @@ def no_fallback_search(search_parameters, provider):
-    for idx in range(len(df)):
-        product_id = df["id"].iloc[idx]
-        if product_id is None:
-            time_string = "id_NA"
-        time_string = df["time"].iloc[idx]
-        if time_string is None:
-            time_string = "time_NA"
-        else:
-            time_string += "Z"
-        cloud_cover_string = df["cloudCover"].iloc[idx]
-        if cloud_cover_string is not None:
-            cloud_cover_string = f"{cloud_cover_string:2.0f}%"
-        else:
-            cloud_cover_string = "cloudCover_NA"
-        product_type_string = df["productType"].iloc[idx]
-        if product_type_string is None:
-            product_type_string = "productType_NA"
-        print(f"{product_id} {time_string} {cloud_cover_string} {product_type_string}")
+    start_date = options["start"]
+    if not options["start"]:
+        delta_days = timedelta(60)
+        start_date = (datetime.fromisoformat(end_date) - delta_days).isoformat()
+    try:
+        start_date = normalize_time(start_date)
+    except Exception as e:
+        gs.debug(e)
+        gs.fatal(_("Could not parse 'start' time."))
 
+    if end_date < start_date:
+        gs.fatal(
+            _(
+                "End Date ({}) can not come before Start Date ({})".format(
+                    end_date, start_date
+                )
+            )
+        )
+    options["start"] = start_date
+    options["end"] = end_date
+
+
+def filter_result(search_result, geometry, **kwargs):
+    """Filter results to comply with options/flags.
+    :param search_result: Search Result to filter
+    :type search_result: class:'eodag.api.search_result.SearchResult'
+
+    :param geometry: WKT String with the geometry to filter with respect to
+    :type geometry: str, optional
+
+    :param kwargs: options/flags from gs.parser, with the crietria that will
+                    be used for filtering.
+    :type kwargs: dict
+
+    :returns: A collection of EO products matching the filters criteria.
+    :rtype: class:'eodag.api.search_result.SearchResult'
+    """
+    prefilter_count = len(search_result)
+    area_relation = kwargs["area_relation"]
+    minimum_overlap = kwargs["minimum_overlap"]
+    cloud_cover = kwargs["clouds"]
+    start_date = kwargs["start"]
+    end_date = kwargs["end"]
+
+    # If neither a geometry is provided as a paraemter
+    # nor a vector map is provided through "options",
+    # then none of the geometry filtering will take place.
+    if geometry is None and (area_relation is not None or minimum_overlap is not None):
+        geometry = get_aoi(kwargs["map"])
+    gs.verbose(_("Filtering results..."))
+
+    if area_relation or minimum_overlap:
+        # Product's geometry intersects with AOI
+        if area_relation == "Intersects":
+            search_result = search_result.filter_overlap(
+                geometry=geometry, intersects=True
+            )
+        # Product's geometry contains the AOI
+        elif area_relation == "Contains":
```

Comment:
I made the minimum_overlap and area_relations options exclusive, that is why it works. But I agree, it would make more sense if I move it to another separate condition.